src/logic/Logic.js

```diff
@@ -578,7 +579,13 @@ class Logic {
     updateRaceModeBannedLocations() {
         _.forEach(potentialBannedLocations, (locations, area) => {
             _.forEach(locations, (location, check) => {
-                const itemLocation = this.getLocation(area, check);
+                let itemLocation;
```

Comment:
```suggestion
        const skyviewCubeLocations = [
            this.cubeList['Goddess Cube in Skyview Spring'],
            this.getLocation('Sky', 'Lumpy Pumpkin - Goddess Chest on the Roof'),
        ];
        skyviewCubeLocations.forEach((loc) => {
            if (loc.settingsNonprogress) {
                return;
            }
            loc.nonprogress = !this.isDungeonRequired('Skyview');
        });
```
github won't let me apply the suggestion to the outer forEach so I just put it here. I really don't think it's a good idea to assume the goddess cube will always be the only extra check of Skyview, so I refactored it to name the 2 locations here directly that are affected